sdk/keyvault/azkeys/azcrypto/crypto_client_test.go

```diff
@@ -0,0 +1,159 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azcrypto
+
+import (
+	"context"
+	"crypto/sha256"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+	"github.com/stretchr/testify/require"
+)
+
+var ctx = context.Background()
+
+func TestMain(m *testing.M) {
+	// Initialize
+	if recording.GetRecordMode() == "record" {
+		vaultUrl := os.Getenv("AZURE_KEYVAULT_URL")
+		err := recording.AddURISanitizer(fakeKvURL, vaultUrl, nil)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Run
+	exitVal := m.Run()
+
+	// cleanup
+
+	os.Exit(exitVal)
+}
+
+func TestNewClient(t *testing.T) {
+	for _, key := range []string{"https://mykeyvault.vault.azure.net/keys/keyabcdef/1234567890", "https://mykeyvault.vault.azure.net/keys/keyabcdef"} {
+		cred := getCredential(t)
+
+		client, err := NewClient(key, cred, nil)
+		require.NoError(t, err)
+
+		require.Equal(t, client.vaultURL, "https://mykeyvault.vault.azure.net/")
+		require.Equal(t, "keyabcdef", client.keyID)
+		if strings.Contains(key, "1234567890") {
+			require.Equal(t, client.keyVersion, "1234567890")
+		} else {
+			require.Equal(t, client.keyVersion, "")
+		}
+	}
+}
+
+func TestClient_Encrypt(t *testing.T) {
```

Comment:
Looks redundant beside `TestClient_Decrypt`, which is more useful because it tests the round trip